torch/linalg/__init__.py

```diff
@@ -140,3 +140,51 @@
     >>> LA.norm(m[0, :, :]), LA.norm(m[1, :, :])
     (tensor(3.7417), tensor(11.2250))
 """)
+
+tensorsolve = _add_docstr(_linalg.linalg_tensorsolve, r"""
+linalg.tensorsolve(input, other, dims=None, *, out=None) -> Tensor
+
+Computes a tensor ``x`` such that ``tensordot(input, x, dims=x.ndim) = other``.
+The resulting tensor ``x`` has the shape equal to ``Q``, where ``Q`` is the shape
+of the sub-tensor of `:attr:`input` consisting of the appropriate number of
+its rightmost dimensions such that ``input.shape = other.shape + Q``.
+Before computing ``x``, dimensions of `:attr:`input` can be moved to the right
+using :attr:`dims` to match the shape requirements.
+
+Supports real and, only on the CPU, complex inputs.
+
+.. note:: If :attr:`input` is not a 'square' tensor, meaning it does not satisfy the requirement
+          ``prod(x.shape) == prod(other.shape)``, then a RuntimeError will be thrown.
+
+Args:
+    input (Tensor): left-hand-side tensor of shape ``other.shape + Q``.
+        ``Q`` is the shape of that sub-tensor of `:attr:`input` consisting of the appropriate
+        number of its rightmost dimensions, and must be such that ``prod(Q) == prod(other.shape)``.
+    other (Tensor): right-hand-side tensor of any shape.
+    dims (Tuple[int]): dimensions of `:attr:`input` to be moved to the right, before solve.
```

Comment:
Saying the dimensions are "moved to right" is a little vague (how far to the right? in what order? what happens to the other dims?). Can this describe the behavior in terms of another function, like movedim? Is it, for example, that each dim is "movedim'd" to the -1 position in the order it appears it the tuple?

Reply:
How about this?
```
    dims (Tuple[int]): dimensions of `:attr:`input` to be moved to the end keeping the order,
                       i.e. ``movedim(input, dims, range(len(dims) - input.ndim, 0)), before solve.
                       If None (default), no moving of dimensions is done.
```